helm/hpcc/templates/_helpers.tpl

```diff
@@ -1045,11 +1045,42 @@ Pass in dict with root, category.  optional name to restrict it to a single name
 
 {{/*
 Create placement related settings
-Pass in dict with placement
+Pass in dict with current placment and new placements dictinary for this entity
+*/}}
+{{- define "hpcc.addPlacementSetting" -}}
+{{- if .me.placement.nodeSelector }}
+{{- $_ := set .new "nodeSelector" (merge (.new.nodeSelector | default dict ) .me.placement.nodeSelector)  }}
+{{- end -}}
+{{- if .me.placement.tolerations }}
+{{- $_ := set .new "tolerations" (concat (.new.tolerations | default list ) .me.placement.tolerations)  }}
+{{- end -}}
+{{- if .me.placement.affinity }}
+{{- $_ := set .new "affinity" .me.placement.affinity  }}
+{{- end -}}
+{{- if .me.placement.schedulerName }}
+{{- $_ := set .new "schedulerName" .me.placement.schedulerName }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Add placement related settings
+Pass in dict with all placements for this entity
 */}}
 {{- define "hpcc.doPlacement" -}}
-{{- if .me.placement }}
-{{ toYaml .me.placement }}
```

Comment:
to generalize/avoid testing each key, you can express as:
```
{{- if len $me -}}
{{ toYaml .me }}
{{- end -}}
```

Reply:
I can't do {{ toYaml .me }} since only need print key, such as "nodeSelector" in the first "nodeSelector" section

Before it was {{ toYaml .me.placement }} but key will be duplicated. For examle,
nodeSelector:
    key1: value1
nodeSelector:
    key2: value2

Reply:
I may be missing something. Now that addPlacementSetting is called multiple times and merges settings into a single nodeSelector, toleraations etc., and doPlacement is called only once. There will be only 1 nodeSelector key generated afaics.

I tried this change and tested with:
```
placements:
- pods: ["all"]
  placement:
    nodeSelector:
      group: "hpcc"
- pods: ["type:dali"]
  placement:
    nodeSelector:
      spot: "false"
```

and it generates (under dali):
```
    spec:
      nodeSelector:
        group: hpcc
        spot: "false"
```

Reply:
Right, that is the limitation. I thought I write it in the placement.md but probably not. I didn't parse the settings before merge since other will need extract each line and get the key. It will be even harder for tolerations since the key can be the same but different handling.  I prefer just add it as limitation since nodeSelector probably will be removed to favor affinity.

Reply:
I add this in the placement.md for nodeSelector:
One key can only be applied once for each item in the "pods" list. Otherwise it will result duplicate settings for the same key.

Reply:
what is wrong with:
```
    spec:
      nodeSelector:
        group: hpcc
        spot: "false"
```

?